### PR TITLE
Check if file exist before opening

### PR DESCRIPTION
**Pull request by MoebiuZ**
When opening a file writing its path directly on the textfield, it wasn't checking if exists (also it wasn't adding it to recent list)

### CODE REVIEW

**Comment by hteso** [comment]
Nice addition the check to verify the file exists. I'm not sure though if we should add the file to the recents list until the user decides to open it. I would add it only once the user accepts and goes to the options dialog.

What is your opinion?

**Comment by MoebiuZ** [comment]
I'm not at the office right now to check, but isn't it already adding it when using Open button?

**Comment by MoebiuZ** [comment]
Checked. This PR fixes exactly that (the recent list code was in the "Select" file dialog, so it was being added before going to "Options" dialog).

I think is misleading the name of the class (should be LoadFileDialog instead NewFileDialog, to not confuse it with the function newfileButton that references the file selection "subdialog". 

Perhaps in a future refactoring.

File: src/newfiledialog.cpp [hunk 1]
```diff
@@ -90,11 +90,22 @@ void NewFileDialog::on_loadFileButton_clicked()
 {
     // Check that there is a file selected
     QString fname = ui->newFileEdit->text();
-    if (fname.isEmpty()) {
+    QFileInfo checkfile(fname);
+    if (!checkfile.exists() && !checkfile.isFile()) {
```
Comment (hteso) [inline src/newfiledialog.cpp]: Shouldn't this be an `||` instead of an `&&`?

Comment (MoebiuZ) [reply src/newfiledialog.cpp]: No. isFile checks the type (file, not dir or symlink)

Comment (hteso) [reply src/newfiledialog.cpp]: Indeed, but both cases separately are also stoppers, not just both of them at the same time. Meaning, if it is not a file or it doesn't exist, in an case, user should select a new one, right?

Comment (MoebiuZ) [reply src/newfiledialog.cpp]: You are right, sorry (in my head I was thinking the oposite case, without the nots)

**Pull request merged**